src/back/spv/block.rs

```diff
@@ -1690,13 +1696,23 @@ impl<'w> BlockContext<'w> {
     pub(super) fn write_block(
         &mut self,
         label_id: Word,
-        statements: &[crate::Statement],
+        naga_block: &crate::Block,
         exit: BlockExit,
         loop_context: LoopContext,
+        debug_info: Option<&DebugInfo>,
     ) -> Result<(), Error> {
         let mut block = Block::new(label_id);
-
+        let statements = naga_block.iter();
+        let mut spans = naga_block.span_iter();
         for statement in statements {
```

Comment:
```suggestion
        for (statement, span) in naga_block.span_iter() {
```

`span_iter` returns both